frontend/src/metabase/App.jsx

```diff
@@ -37,9 +37,22 @@ const getErrorComponent = ({ status, data, context }) => {
 
 @connect(mapStateToProps)
 export default class App extends Component {
+  state = {
+    hasError: false,
+  };
+
+  componentDidCatch(error, info) {
+    console.error("Error caught in <App>", error, info);
+    this.setState({ hasError: true });
+  }
+
   render() {
     const { children, location, errorPage } = this.props;
 
+    if (this.state.hasError) {
+      return <div>😢</div>;
```

Comment:
Under what conditions will `hasError` trigger?

Reply:
`componentDidCatch` is a React 16 feature, so never until we upgrade. I could remove this code until then, or leave it in as a placeholder.